zoom_map.py

```diff
@@ -79,26 +88,57 @@ def add_level(level):
                 [("<Button-4>", partial(self._map_zoom, -1)),
                  ("<Button-5>", partial(self._map_zoom,  1)),
                  ("<Button-1>", self._on_click),
-                 ("<B1-Motion>", self._on_drag)]]
+                 ("<B1-Motion>", self._on_drag),
+                 ("<ButtonRelease-1>", self._on_unclick)]]
 
     def _set_image(self):
-        self._tk_image = self.create_image(0, 0, anchor=tk.NW,
-                                           image=self.image)
+        if self._tk_image is not None:
```

Comment:
I think this is a stylistic difference. I'd prefer to check None explicitly, since an "existence" check really checks whether it's falsey, and then the reader needs to wonder if it is None or False or the empty list, or the empty string, or whether empty numpy arrays are falsey, etc. I prefer explicitly saying "this might be None." and yet, that's more verbose than it needs to be, and does seem overly specific. I'll change it if you want, but if left to my own devices, I prefer the explicitness of specifying None.